Mlem/App/Views/Shared/ContentLoader.swift

```diff
@@ -0,0 +1,37 @@
+//
+//  ContentLoader.swift
+//  Mlem
+//
+//  Created by Eric Andrews on 2024-05-13.
+//
+
+import Foundation
+import MlemMiddleware
+import SwiftUI
+
+struct ContentLoader<Content: View, Model: Upgradable>: View {
+    let model: any Upgradable
+    
+    var content: (Model.Upgraded) -> Content
+    
+    init(model: Model, content: @escaping (Model.Upgraded) -> Content) {
+        self.model = model
+        self.content = content
+    }
+    
+    var body: some View {
+        if let upgradedModel = model.upgraded as? Model.Upgraded {
+            content(upgradedModel)
+        } else {
+            ProgressView()
+                .task {
+                    print("upgrading...")
+                    do {
+                        try await model.upgrade()
+                    } catch {
+                        print(error)
+                    }
+                }
+        }
+    }
```

Comment:
This logic works great when upgrading a `Stub` all the way to `Upgraded`. But what about if we're upgrading a mid-tier model? In those cases, we'd want to display what information we *do* have whilst the upgrade is in progress. For example, we'd be able to display all of the post information using just a `Post1`, except for the status of interactions. Should `ContentLoader` have an option to display the `content` with any non-stub model, or is the intent that we wouldn't be using `ContentLoader` in cases where we need to do this?